examples/all-components/src/ds/mod.rs

```diff
@@ -0,0 +1,38 @@
+mod components;
+mod elements;
+mod theme;
+
+pub use components::*;
+pub use elements::*;
+pub use theme::*;
+
+use freya::prelude::*;
+
+#[component]
+pub fn DsGui(children: Element) -> Element {
+    let theme_signal = use_init_default_theme();
+
+    rsx!(
+        rect{
+            main_align: "center",
+            cross_align: "center",
+            height: "100%",
+            width: "100%",
+            rect {
+                height: "80%",
+                width: "100%",
+                ThemeProvider {
+                    theme: theme_signal()
+                    {children}
+                }
+            }
+            rect{
+                height: "20%",
+                width: "100%",
+                ThemeEditor{
+                    theme: theme_signal
```

Comment:
You don't really need to pass the theme signal as prop, instead you can just call `use_theme` in all the components you need the signal.